vessel-runtime/src/main/java/com/textnow/android/vessel/VesselCacheImpl.kt

```diff
@@ -0,0 +1,84 @@
+package com.textnow.android.vessel
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+
+/**
+ * [DefaultVesselCache] has no max size and no eviction policy. Data is stored
+ * in memory until the process is killed, or [clear] is called.
+ */
+open class DefaultVesselCache : VesselCache {
+    private val hashMap = ConcurrentHashMap<String, Any>()
+
+    override val size: Int
+        get() = hashMap.size
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : Any> get(key: String): T? {
+        return hashMap[key] as? T?
+    }
+
+    override fun <T : Any> set(key: String, value: T) {
+        hashMap[key] = value
+    }
+
+    override fun remove(key: String) {
+        hashMap.remove(key)
+    }
+
+    override fun clear() {
+        hashMap.clear()
+    }
+}
+
+/**
+ * [LRUVesselCache] has a max size of [capacity]. Adding a new key when the cache is full
+ * results in the least recently used key to be evicted. Data is stored
+ * in memory until the process is killed, or [clear] is called.
+ */
+open class LRUVesselCache(
```

Comment:
Nit: According to the [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names), this should actually be `LruVesselCache`